libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -194,13 +195,7 @@ static bool _validatePublishPayload( const IotMqttPublishInfo_t * pPublishInfo,
     /* This parameter is not used when logging is disabled. */
     ( void ) pPublishTypeDescription;
 
-    if( pPublishInfo == NULL )
-    {
-        IotLogError( "Publish information cannot be NULL." );
-
-        status = false;
-    }
-    else if( pPublishInfo->payloadLength != 0U )
+    if( pPublishInfo->payloadLength != 0U )
```

Comment:
Since the `else if` was turned into an `if`, it seems the `else` block at line 220 could be removed.